### ISSUE TITLE
All ChromatiCraft progression lost

### ISSUE THREAD

**Issue by Turmfalke2** [ReikaKalseki/Reika_Mods_Issues#1706]
hi,
I just lost all my ChromatiCraft progression.

I was in the chroma dimension, travelling through a lumen stream towards the farlands. While doing that I was reading some entries in the ChromatiCraft book, so at this point I definitely still had my progression. After leaving the stream at the end I found that world below wouldn't load so I restarted minecraft. I then travelled a few thousands blocks outwards until I noticed that I was missing progression to see the chroma plants, at this point I checked my book and found all progression to be missing.

To recap, minecraft never crashed, but it was apparently lost when I restarted minecraft due to the world not loading.

Which files do I need to restore from a backup?

E:
Restarted minecraft again afterwards and got the following error:

https://pastebin.com/gvgVw7R1

However I still have no progression.

**Comment by TsumiHokiro** [comment]
It's been 4 days already so Idk if you managed to find the file, but it's in the folder Chromaticraft_Data/ProgressionBackup/[ID].dat
Do notice this is just a copy of the ForgeData inside the normal playerdata [or the player part of the level.dat] and as such does not have other info such as items/position/etc.

**Comment by Turmfalke2** [comment]
> It's been 4 days already so Idk if you managed to find the file,

I did not, but I just had a closer look at the files you mentioned. The Chromaticraft_Data/ProgressionBackup/[ID].dat looks fine, however I can't find the same data in the level.dat

**Comment by TsumiHokiro** [comment]
In case it is single player world, the correct path you want is "level.dat/Player/ForgeData/PlayerPersisted". Since in single player there is only 1 player, there's no identification of who the player is, therefore the playerdata folder has only a copy of this.
In case you did lose your progress, or the path was not automatically created, creating a new one should be no problem (At least it did work for me when I lost my progression [and everything else...] and stitched back this file back into level.dat) **as long as you preserve the structure**. Do note that PlayerPersisted has a lot more info to it than just Chromaticraft and you might just want to copy the info relevant to Chromaticraft.
Edit: Rephrasing.

**Comment by Turmfalke2** [comment]
Right, the ChromatiCraft Data in PlayerPersisted is pretty empty. How does that even happen? I would assumed that the level.dat file gets written in one go, so how is something in the middle of it missing?

How did you stitch the file together again? Do you have a library for manipulating nbt?

E: found something that looks promising https://hackage.haskell.org/package/nbt

**Comment by ReikaKalseki** [comment]
There are many, many public NBT editors.

**Comment by TsumiHokiro** [comment]
I mostly use NBTExplorer to do any nbt editing. Not really user friendly but it does get the job done. You could try looking for NBT editors, there are a few out there.
As for how does that happen... well... many things could happen, that's what would probably be called a bug. If you could find a way to reproduce your loss of data, it could help. On the other hand, it could also be many kinds of software/hardware failure, not directly related to Chromaticraft in the least.
